### DISCUSSION TITLE
Validate format of Markdown files

### DISCUSSION THREAD

**Discussion by daniel-zullo-frequenz** [frequenz-floss/frequenz-repo-config-python#23] [Ideas]
We could add a linter or static analysis tool to the CI to ensure markdown files are consistent and properly formatted. This might also avoid further discussion within the team about different ways of formatting Markdown files.

This is just a lint tool that I'm personally using with VS code https://github.com/DavidAnson/markdownlint but there might be other ones more suitable to integrate in the CI.

This discussion started in a PR comment https://github.com/frequenz-floss/frequenz-channels-python/pull/87#discussion_r1190790413

fyi @frequenz-floss/python-sdk-team

**Comment by Marenz** [comment]
markdownlint sounds nice, but it might be preferable to use something python based if we have the option.

**Comment by Marenz** [comment]
possibly https://pypi.org/project/pymarkdownlnt/ ?

**Comment by leandro-lucarella-frequenz** [comment]
I haven't looked at the links yet, but just to have it in mind, we I think we should prioritize the following:

* Supports github flavored markdown
* Can be easily integrated to the most common editors/IDEs (so we don't need to find out the linting fails only after creating a PR)
* Is a maintained tool

**Comment by leandro-lucarella-frequenz** [comment]
BTW, comparisons between these 2 tools (and any other **relevant** alternative), paying particular attention to the list above, are very welcome!
